ufo2fontir/src/source.rs

```diff
@@ -435,6 +435,37 @@ fn glyph_order(
     Ok(glyph_order)
 }
 
+fn units_per_em(
+    designspace: &DesignSpaceDocument,
+    designspace_dir: &Path,
+) -> Result<u16, WorkError> {
+    let mut upems: HashSet<u16> = HashSet::new();
+    for source in designspace.sources.iter() {
+        let ufo_dir = designspace_dir.join(&source.filename);
+        if !ufo_dir.join("fontinfo.plist").exists() {
+            continue;
+        }
+        let fontinfo = load_plist(&ufo_dir, "fontinfo.plist")?;
+        let Some(plist::Value::Integer(upem)) = fontinfo.get("unitsPerEm") else {
```

Comment:
for whatever reason, UFO allows `unitsPerEm` to be "non-negative integer _or float_" (emphasis mine)... I have never seen a float upem in the wild, and can't even imagine one (a global _grid_ to be used as reference for the rest of font unit coordinates, is the least float-y thing in the world, IMHO).
Anyway, [ufo2ft applies an otRound to it before setting in the head table](https://github.com/googlefonts/ufo2ft/blob/5b22f201b4650ab9ef1169357702a732e3127a5e/Lib/ufo2ft/outlineCompiler.py#L257), we might as well do that here

https://unifiedfontobject.org/versions/ufo3/fontinfo.plist/#generic-dimension-information

(At least Glyphs.app types it as a integer so we are fine in glyphs2fontir).

Reply:
Really?! That's amazing. Good catch, will fix.

Reply:
Fixed.

Reply:
And added a test to prove it.